modules/stochastic_tools/src/actions/StochasticResultsAction.C

```diff
@@ -0,0 +1,62 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "StochasticResultsAction.h"
+#include "ActionWarehouse.h"
+#include "ActionFactory.h"
+#include "Transfer.h"
+#include "SetupMeshAction.h"
+#include "SamplerPostprocessorTransfer.h"
+#include "StochasticResults.h"
+
+registerMooseAction("StochasticToolsApp",
+                    StochasticResultsAction,
+                    "declare_stochastic_results_vectors");
+
+InputParameters
+StochasticResultsAction::validParams()
+{
+  InputParameters params = Action::validParams();
+  params.addClassDescription("Action for performing initialization of StochasticResults vectors "
+                             "based on SamplerPostprocessorTransfer.");
+  return params;
+}
+
+StochasticResultsAction::StochasticResultsAction(InputParameters params) : Action(params) {}
+
+void
+StochasticResultsAction::act()
+{
+  if (_current_task == "declare_stochastic_results_vectors")
+  {
+    for (std::shared_ptr<Transfer> & transfer_ptr :
+         _problem->getTransfers(Transfer::DIRECTION::FROM_MULTIAPP))
+    {
+      std::shared_ptr<SamplerPostprocessorTransfer> ptr =
+          std::dynamic_pointer_cast<SamplerPostprocessorTransfer>(transfer_ptr);
+      if (ptr != nullptr)
+      {
+        const VectorPostprocessorName & result_name =
+            ptr->getParam<VectorPostprocessorName>("to_vector_postprocessor");
+        const std::vector<VectorPostprocessorName> vpp_names = ptr->vectorNames();
+
+        // Get the StochasticResults storage object, get it by base class to allow for better
+        // type check error message
+        auto & uo = _problem->getUserObject<UserObject>(result_name);
```

Comment:
`auto` here? It would be nice to be consistent within the same method. I would say anywhere in this method where you have a type on the RHS you might as well use `auto` on the LHS